model/host/host.go

```diff
@@ -595,10 +595,22 @@ func (h *Host) IdleTime() time.Duration {
 		}
 	}
 
-	// The host isn't ready to run tasks yet so it's not idle.
+	// The host isn't ready to run tasks yet, so it's not idle.
 	return 0
 }
 
+// ShouldNotifyRunningSpawnHostIdle returns true if the running spawn host has been idle long enough to notify the user.
+func (h *Host) ShouldNotifyRunningSpawnHostIdle() bool {
```

Comment:
@Kimchelly this might not interact well with the sleep scheduler since a "running" host could be marked as stopped at the expiration time; should this return take into account some metadata that denotes it as using the sleep scheduler? Or should we just bail on notifying for particular statuses and just generally notify at the three month mark?

Reply:
I don't think communication time is something we can use to determine idleness because that indicates the last time the app server talked to the host. If the user's just kept it on the whole time and done regular stuff like SSH, Evergreen would have no way of knowing.

I think running hosts should be left alone. We would have to ask the user directly about their usage to know if it's really idle.

Reply:
If the user _hasn't_ done regular stuff like ssh, then it would certainly have a stale communication time, right? I think I'm trying to catch the case detailed in the ticket, where users have truly forgotten about their hosts (even running hosts -- there are 161 hosts right now that match this query:
```
 db.hosts.count({"status":"running", "last_communication": {"$lte": ISODate('2023-01-01T00:00:00.000Z')}})
```

Reply:
Ah okay I see what you're saying; that's fair, I'll change this to just alert on stopped hosts that are three months idle.

Reply:
I think I still have an open question though -- is there a good way to know if the user has it permanently stopped, as opposed to it just being at the stopped part of the sleep schedule? I'm thinking its ShouldKeepOff -- have we started setting that field for currently stopped hosts yet, or is that something I can do as part of this PR?

Reply:
If they're on the sleep schedule, their host stops/start at least once per week, so it's not really relevant. If they're exempt (for whatever reason), the question is still valid if they're really using the host when it's stopped.

Reply:
I don't totally understand the response above but I think what I'm asking is -- we should only notify people if there host has actually been stopped for a long time because they stopped it manually, which I'm assuming prevents the sleep scheduler from setting it back to running.

I just tried stopping a host and it doesn't look like ShouldKeepOff was set to true, so casing on this doesn't make sense. Is that eventually the goal for this variable, or do we not have a plan to keep manually stopped hosts stopped indefinitely? 

Should I remove checking ShouldKeepOff for now and just make a note to whatever ticket we're addressing that in?

Reply:
Requesting you directly since I want to make sure this is written so it works both now and after the sleep scheduler is turned on. Feel free to just DM if that's easier.

Reply:
Basically, the sleep schedule should have no effect on this notification, because a long-stopped host is a long-stopped host no matter why it's stopped. ShouldKeepOff has no effect because the UI doesn't exist for it yet, but even if it did, that's just a way to keep the host off for as long as you want. Keeping it off as long as you want is basically the same as an unused host, which is what the notification is for.